ccm_web/server/src/config.ts

```diff
@@ -119,7 +145,10 @@ export function validateConfig (): Config {
     canvas = {
       instanceURL: validate<string>('CANVAS_INSTANCE_URL', env.CANVAS_INSTANCE_URL, isString, [isNotEmpty]),
       apiClientId: validate<string>('CANVAS_API_CLIENT_ID', env.CANVAS_API_CLIENT_ID, isString, [isNotEmpty]),
-      apiSecret: validate<string>('CANVAS_API_SECRET', env.CANVAS_API_SECRET, isString, [isNotEmpty])
+      apiSecret: validate<string>('CANVAS_API_SECRET', env.CANVAS_API_SECRET, isString, [isNotEmpty]),
+      customCanvasRoleData: validate<CustomCanvasRoleData>(
+        'CANVAS_CUSTOM_ROLES', prepObjectFromJSON(env.CANVAS_CUSTOM_ROLES), isCustomCanvasRoles, [], { Assistant: 34, Librarian: 21 }
```

Comment:
I noticed that the fallback just looks like `[object Object]` in the log message. This isn't a significant problem, but we could always `JSON.stringify` on the fallback, since that method also handles primitive values. Seems like it would fail with complicated objects or `bigint`, but we are not using either currently. https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/JSON/stringify

Reply:
I'm okay with doing nothing here if it seems on the risky side.